contrib/nydus-snapshotter/pkg/filesystem/nydus/fs.go

```diff
@@ -373,7 +377,10 @@ func (fs *filesystem) createSharedDaemon(snapshotID string, imageID string) (*da
 		d            *daemon.Daemon
 		err          error
 	)
-
+	d, _ = fs.manager.GetBySnapshotID(snapshotID)
```

Comment:
Though, seems it is also possible that a race window exists where two goroutines are creating daemons with those directories created.